src/app/sign-up/page.tsx

```diff
@@ -161,6 +163,178 @@ function CustomerSignupForm() {
             Password
           </Text>
           <TextField.Root
+            required
+            type="password"
+            name="password"
+            placeholder="********"
+            mt="2"
+          />
+        </Box>
+
+        <Flex direction="row" width="100%" justify="end" gap="4">
+          <Link href="sign-in">
+            <Button type="button" color="indigo" variant="soft">
+              Already have an account?
+            </Button>
+          </Link>
+          <Button color="indigo" variant="solid">
+            Sign Up
+          </Button>
+        </Flex>
+      </Flex>
+    </form>
+  )
+}
+
+function ServiceProviderSignupForm() {
+  const signUp = async (formData: FormData) => {
+    const email = formData.get('email') as string
+    const password = formData.get('password') as string
+    const preferredName = formData.get('preferred-name') as string
+    const fullname = formData.get('fullname') as string
+    const companyName = formData.get('company-name') as string
+    const abn = formData.get('abn') as string
+    const acn = formData.get('acn') as string
+    const industry = formData.get('industry') as string
+
+    const supabase = createClient()
+
+    const { error } = await supabase.auth.signUp({
+      email,
+      password,
+      options: {
+        data: {
+          role: 'service-provider',
+          onboarded: false,
```

Comment:
For Service Providers, we need to create a user via Stripe's Accounts product. This will return an onboarding link, where the provider will have to confirm their identity and ownership of the company.

We'll need to store the onboarding link along with the onboarding status. Think there might be a webhook involved with this once onboarding has completed.

Reply:
We'll also need a way to flag that the Provider's Profile has been setup. Will have a think about having one flag or separate flags for the Provider's metadata.